### ISSUE TITLE
[UX] Change "View mode" to "Display mode" to avoid confusion with views module

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2376]
We have a ton of places in the UI were we refer to an entity `View mode`. In https://github.com/backdrop/backdrop-issues/issues/315 we wanted to change the name of this setting to "Display mode" or "Custom display mode" or simply "Display" to avoid any confusion (or unwanted association) with the Views module.

I started working on a PR to change the terminology in the user interface, and noticed that we use `view modes` so heavily in code, that I wondered if the change here was worth it. I'm going to file a PR so everyone can see how much change this would be, but it's a lot.

Pro: less confusing UI.
Con: inconsistency between UI and code. 

---
PR: https://github.com/backdrop/backdrop/pull/1993

**Comment by jenlampton** [comment]
Marking as NW because the provided PR is on top of @docwilmot's work in https://github.com/backdrop/backdrop/pull/1645

**Comment by jenlampton** [comment]
Oh, @klonos already had an issue for this :)  Closing this one in favor of his. https://github.com/backdrop/backdrop-issues/issues/2342

**Comment by jenlampton** [comment]
Actually, re-opening since that issue seems to be quite specific and includes other UI changes too.

**Comment by klonos** [comment]
> Con: inconsistency between UI and code.

We can create a follow-up issue to change the code part (in 2.x if it is too much change) or we can do it now and add wrapper functions that make the old code still work.

**Comment by jenlampton** [comment]
This isn't the kind of code change where wrapper functions are going to be any help. There are keys in a gazillion arrays for `$entity['view_mode']`. 

This is the kind of code change that isn't worth the cost. Contrib developers are going to be very angry with us if they have to update hundreds of lines of code just to change array keys.

**Comment by quicksketch** [comment]
@jenlampton Now that you've had your head in the Entity View Modes issue for weeks, how do you feel about this issue? I think the terminology difference in this case is something inavoidable in 1.x.

**Comment by jenlampton** [comment]
Now that the manage display page is renamed "Manage displays" it makes me want to rename view modes to "Display modes" even more. Let's leave the DX/UX inconsistency and see if we can increase UX for 1.6.  We can increase DX in 2.0 :)

**Comment by quicksketch** [comment]
As this needs work and makes some significant changes to the UI, I'm moving this out of the 1.6.0 queue.

**Comment by quicksketch** [comment]
Descoping for 1.7.0

**Comment by jenlampton** [comment]
Even though we still have 15 days?  I was leaving this one till after today :)

**Comment by jenlampton** [comment]
Descoping for 1.7.0.

**Comment by jenlampton** [comment]
Pushed a fresh PR that should apply cleanly.

**Comment by olafgrabienski** [comment]
I like the new terminology, and the PR looks quite good in the interface but "some checks were not successful".

I'm not sure about the action link "Add Display mode". Before it was "Add view mode" ("view" not starting with a capital). Compare also the new "Each Page **D**isplay mode can be configured differently" vs. the former "Each Page **v**iew mode can be configured ..."

**Comment by jenlampton** [comment]
@olafgrabienski where are you looking that you see the action link "Add Display mode" with a capital D? I've fixed up all other places the capitalization seemed off, but was unable to find the action link.

**Comment by docwilmot** [comment]
The PR looks technically good. 
But I thought this issue was only to change end-user-facing texts and labels; as it is we're changing code comments as well, leading to mismatches between comments and variables/array keys, like:
```
 // display modes for nodes are defined here and not in search.module.
      $return['node']['view modes'] += array(
```
A newbie coder may be confused.

**Comment by olafgrabienski** [comment]
@jenlampton I took a screenshot with the action link in the sandbox site (see below), *then* cleared all caches, and voilà: the capitalization *disappeared*, so everything seems to be fine!

Btw, off topic: have a look at the dropbuttons on the screenshot (around 930px screen width).

![screen-backdrop-pr-1993-display-modes](https://user-images.githubusercontent.com/14188307/30476433-5241046c-9a0a-11e7-8614-27f9672ab75a.png)

**Comment by opi** [comment]
[off-topic] @olafgrabienski dropbutton overflow is fixed by https://github.com/backdrop/backdrop-issues/issues/2806#issuecomment-329634013

**Comment by olafgrabienski** [comment]
@opi Good to know! (Wasn't sure if that fix was supposed to be already in this PR here.)

**Comment by jenlampton** [comment]
@docwilmot the idea was to avoid changing the keys in arrays, so things like the below should not change.
```php
 $return['node']['view modes'] += array(
``` 

If however, there is nothing in  Backdrop called a "view mode" a code comment telling a newbie coder about a thing that doesn't exist is going to be very confusing. The code comment should tie the code (with these mysterious keys) to something familiar, so we should be using the terminology someone would see in the interface: "display mode". This will help newcomers make sense of our legacy array keys.

**Comment by docwilmot** [comment]
@jenlampton I dont suppose with a change like this we can avoid confusing at least some people, so looks good to me. RTBC.

**Comment by quicksketch** [comment]
Thanks @docwilmot and @olafgrabienski for the reviewing! Yeah, this won't be the first time we have disconnects between front-end and APIs, e.g. "node" vs. "content" or Views "arguments" vs. "contextual filters".

I think the documentation updates are largely unnecessary, using the examples of "node" and "arguments": we still refer to them as those things in code and docblocks all the time, and they're used interchangeably with their alternatives. But the docs updates are all done in the PR already, so I don't see any reason to roll them all back.

I only found 2 minor issues in the documentation changes: https://github.com/backdrop/backdrop/pull/1993#pullrequestreview-63194562. After that, this looks good to go to me.

**Comment by jenlampton** [comment]
Pushed a new PR with the two fixes.

**Comment by quicksketch** [comment]
Thanks! Looks great. Merged backdrop/backdrop#1993 into 1.x for 1.8.0.
